### ISSUE TITLE
Cluster.Tools Pub/Sub removes active subscriptions

### ISSUE THREAD

**Issue by kantora** [akkadotnet/akka.net#2226]
The history of the problem:
https://github.com/akkadotnet/akka.net/issues/2221
https://github.com/akkadotnet/akka.net/pull/2222
https://github.com/akkadotnet/akka.net/pull/2225

In my case, I have multiple subscribers and rare publishes (I use only `Publish`/`Subscribe`). But almost always (not every time) in 120s after node start (this is the default prune timeout) `DistributedPubSubMediator.HandlePrune` is called and removes everything from `_registry` (I've fixed the obvious bugs, so now it does not throw exceptions and just removes what it was intended to remove).

**Comment by kantora** [comment]
@alexvaluyskiy emmm... the problem was not solved, I've just removed the exception, but subscriptions are still removed in 120s after node start...

**Comment by alexvaluyskiy** [comment]
@kantora I've found something. We are using Tick in NextVersion()
https://github.com/akkadotnet/akka.net/blob/dev/src/contrib/cluster/Akka.Cluster.Tools/PublishSubscribe/DistributedPubSubMediator.cs#L590

But we are comparing with Milliseconds in HandlePrune();
https://github.com/akkadotnet/akka.net/blob/dev/src/contrib/cluster/Akka.Cluster.Tools/PublishSubscribe/DistributedPubSubMediator.cs#L531

Scala version uses Milliseconds in the both places:
https://github.com/akka/akka/blob/master/akka-cluster-tools/src/main/scala/akka/cluster/pubsub/DistributedPubSubMediator.scala#L509
https://github.com/akka/akka/blob/master/akka-cluster-tools/src/main/scala/akka/cluster/pubsub/DistributedPubSubMediator.scala#L816

**Comment by alexvaluyskiy** [comment]
@kantora could you check it?

**Comment by kantora** [comment]
@alexvaluyskiy I'll try to do it in the beginning of the next week.

**Comment by alexvaluyskiy** [comment]
@kantora The fix has been merged. I tried it with my example, and it worked fine. You should try on your example too.

**Comment by kantora** [comment]
@alexvaluyskiy as I see from the commit, you left the problem with the midnight (after the midnight all new versions will be obsolete).
I suppose that `var current = DateTime.UtcNow.Ticks / 10000;` is much better.

**Comment by alexvaluyskiy** [comment]
Right now we have the same implementation, as Akka Scala has. If you think that this implementation is not working correctly, you could create a test where you reproduce a bug. Or write about this problem here
https://github.com/akka/akka/issues/21100

**Comment by kantora** [comment]
@alexvaluyskiy Scala implementation is like this:
`val current = System.currentTimeMillis`
this is the number of ms from the beginning of the universe (or 01.01.1970 as I remember) or standard timestamp.
`DateTime.UtcNow.TimeOfDay.TotalMilliseconds` is the number of ms from the start of the day (and it will reset every UTC midnight).
According to the msdn, Tick is 100ns or 1ms/10000. `DateTime.UtcNow.Ticks / 10000` should give you the correct timetamp as it is made in scala version.

**Comment by alexvaluyskiy** [comment]
Or this
```csharp
        private static double GetCurrentMilliseconds()
        {
            DateTime initialDate = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
            TimeSpan javaSpan = DateTime.UtcNow - initialDate;
            return javaSpan.TotalMilliseconds;
        }
```

**Comment by kantora** [comment]
Yeah, sure :)

**Comment by kantora** [comment]
@alexvaluyskiy I've checked the `Ticks / 10000` variant and it seems to work fine. `GetCurrentMilliseconds` should also work well.

**Comment by alexvaluyskiy** [comment]
My method returns the same value as Java's `System.currentTimeMillis`, but `Ticks / 10000` returns 40x times bigger value. We can't use it in the comparsion

**Comment by kantora** [comment]
That's because of different Windows and Unix  beginning of the universe understanding. In Unix it is 01.01.1970, and in Windows it is 01.01.0001. But if you divide ticks by 10000 you still will get the ms and this is still plausible for comparison. 

P.S. I am not intending on `Ticks`, just saying that both variants are correct, but `GetCurrentMilliseconds` will get the same values as Scala version does.

**Comment by Horusiath** [comment]
I think, it's not important if we use ticks (we should because, it's standard for .NET) or millis. What's important here, is to keep those values consistent.

**Comment by Aaronontheweb** [comment]
Closed by #2269
